modules/apps/web-experience/export-import/export-import-resources-importer/src/main/java/com/liferay/exportimport/resources/importer/internal/messaging/ResourcesImporterHotDeployMessageListener.java

```diff
@@ -73,6 +78,16 @@
 
 	@Activate
 	protected void activate(final BundleContext bundleContext) {
+		_bundleContext = bundleContext;
+
+		_messageQueue = new LinkedList<>();
+
+		_serviceTracker = ServiceTrackerFactory.open(
```

Comment:
I think this can be solved. This is no different than many other scenarios we already have. What you need to do is provide a way to discover, or declare the dependencies that the resources declared actually need. this information should already be in the data since you are referencing models and portlets. All we need to do exactly like we do with service builder and create a dynamic component which waits passively for all those dependencies to be available before actually completing the work. We have at least 3-4 examples of doing this.